### PR TITLE
Update V8 to 8.9.255.25

### PR DESCRIPTION
**Pull request by dahlstrom-g**
b/156155426

### CODE REVIEW

File: cobalt/script/v8c/v8c_global_environment.cc [hunk 1]
```diff
@@ -394,24 +391,16 @@ void V8cGlobalEnvironment::MessageHandler(v8::Local<v8::Message> message,
   v8::Isolate* isolate = v8::Isolate::GetCurrent();
   V8cGlobalEnvironment* global_environment =
       V8cGlobalEnvironment::GetFromIsolate(isolate);
-  if (isolate->GetEnteredContext().IsEmpty()) {
-    return;
-  }
```
Comment (johnxwork) [inline cobalt/script/v8c/v8c_global_environment.cc:399]: I think this is to support some testing scenarios where there is no global object.

File: cobalt/script/v8c/v8c_global_environment.cc [hunk 2]
```diff
@@ -97,9 +97,6 @@ V8cGlobalEnvironment::V8cGlobalEnvironment(v8::Isolate* isolate)
   isolate_->SetData(kIsolateDataIndex, this);
   DCHECK(isolate_->GetData(kIsolateDataIndex) == this);
 
-  isolate_->SetAllowCodeGenerationFromStringsCallback(
-      AllowCodeGenerationFromStringsCallback);
```
Comment (johnxwork) [inline cobalt/script/v8c/v8c_global_environment.cc:101]: I think this gives us an event to handle eval() related web spec. See https://source.corp.google.com/h/lbshell-internal/cobalt_src/+/COBALT:cobalt/script/v8c/v8c_global_environment.cc;drc=7a6537c86df4096546b2fcec40078e431cd0b6d8;l=374

File: cobalt/script/v8c/v8c_global_environment.cc [hunk 3]
```diff
@@ -394,24 +391,16 @@ void V8cGlobalEnvironment::MessageHandler(v8::Local<v8::Message> message,
...
   }
 
-  v8::Local<v8::Context> context = isolate->GetEnteredContext();
   ErrorReport error_report;
   error_report.message = *v8::String::Utf8Value(isolate, message->Get());
   error_report.filename =
       ToStringOrNull(isolate, message->GetScriptResourceName());
   int line_number = 0;
   int column_number = 0;
-  if (message->GetLineNumber(context).To(&line_number) &&
```
Comment (johnxwork) [inline cobalt/script/v8c/v8c_global_environment.cc:411]: This part is just collecting script error details when error happens.

File: cobalt/script/v8c/v8c_global_environment.cc (hunk 2 above)
Comment (dahlstrom-g) [reply cobalt/script/v8c/v8c_global_environment.cc:101]: I should have looked before asking, but I found in revision history:

> Use Isolate::SetModifyCodeGenerationFromStringsCallback instead.
> See http://crbug.com/v8/10096.

File: cobalt/script/v8c/v8c_global_environment.cc (hunk 1 above)
Comment (dahlstrom-g) [reply cobalt/script/v8c/v8c_global_environment.cc:399]: And for this:

> Use GetEnteredOrMicrotaskContext().

**Review by jellefoks — APPROVED** [review]

**Review by jellefoks — APPROVED** [review]

**Pull request merged**